design/legacy-block-storage.md

```diff
@@ -0,0 +1,91 @@
+# Legacy Block Storage with Rook
+
+
+## Overview
+
+### Value of legacy block storage support to Rook
+
+* In order to completely integrate with Kubernetes and utilize Rook's autonomous healing, scaling, and managing features, backend storage systems need to be software-based, distributed (clustered), and capable of being containerized. 
+Not all modern storage systems meet these requirements.
+* Many expensive and proprietary appliance storage silos are still providing services for Enterprise IT infrastructures.
+* Many companies only trust well-certified and time-proven storage solutions for their mission critical applications
+
+For transitioning into existing enterprise IT infrastructures, Rook will have a much easier time with support for legacy block storage systems.
+
+
+### Storage Resource Normalization
+
+The OpenStack Block Storage service (Cinder) provides a standardized architecture to aggregate the features and management of all kinds of block storage systems.
+It is the first service to have a standardized interface that is widely supported by most storage vendors including Dell EMC, IBM, NetApp, Hitachi, HPE, Pure Storage, VMware, etc.
+
+The major functions of the Cinder volume driver include:
```

Comment:
Is the intention that the Rook-Legacy-Block backend support all of these major functions?

Reply:
rook-legacy-block plans to support following at beginning which are needed for CSI.

Create/Delete Volume
Attach/Detach Volume
Create/Delete Snapshot
Create Volume from Snapshot
Mount/Unmount Volume